File: Java/perfect.java
import java.util.Scanner;
public class perfect
{
public static void main(String[]args)
{

Scanner sc = new Scanner(System.in);
System.out.println("Enter Number:");
int n=sc.nextInt();
int sum=0, i = 1;
while (i<n)
{
if (n%i==0)
{
sum =sum+i;
}
i++;
}
if (sum == n)
{
System.out.println(" perfect number ");
}
else
{
System.out.println("not perfect number");
}
}
}

File: DotNetTechnology/perfect.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace lab2
{
    internal class perfect
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Enter the number to check perfect number or not:");
            int n = int.Parse(Console.ReadLine());
            int sum = 0;
            for (int i = 1; i < n; i++)
            {
                if (n % i == 0)
                    sum = sum + i;

            }
            if (sum == n)
                Console.WriteLine(n+" is the perfect number.");
            else
                Console.WriteLine(n+" is not the Perfect number.");
            Console.ReadKey();

        }
    }
}
